src/shared/util.js

```diff
@@ -907,58 +903,37 @@ var PageViewport = (function PageViewportClosure() {
 
     // creating transform to convert pdf coordinate system to the normal
     // canvas like coordinates taking in account scale and rotation
-    var centerX = (viewBox[2] + viewBox[0]) / 2;
-    var centerY = (viewBox[3] + viewBox[1]) / 2;
-    var rotateA, rotateB, rotateC, rotateD;
     rotation = rotation % 360;
-    rotation = rotation < 0 ? rotation + 360 : rotation;
-    switch (rotation) {
-      case 180:
-        rotateA = -1; rotateB = 0; rotateC = 0; rotateD = 1;
-        break;
-      case 90:
-        rotateA = 0; rotateB = 1; rotateC = 1; rotateD = 0;
-        break;
-      case 270:
-        rotateA = 0; rotateB = -1; rotateC = -1; rotateD = 0;
-        break;
-      // case 0:
-      default:
-        rotateA = 1; rotateB = 0; rotateC = 0; rotateD = -1;
-        break;
+    if (rotation < 0) {
+      rotation = rotation + 360;
     }
-
-    if (dontFlip) {
-      rotateC = -rotateC; rotateD = -rotateD;
-    }
-
-    var offsetCanvasX, offsetCanvasY;
+    var sideways = rotation % 180;
+    var a = rotation < 180 ? 1 : -1;
```

Comment:
You could also write down the `transform` matrices explicitely, which would further reduce the number of arithmetic operations: 
```
...
    } else {
      width = scale * Math.abs(viewBox[2] - viewBox[0]);
      height = scale * Math.abs(viewBox[3] - viewBox[1]);
    }

    // creating transform for the following operations:
    // translate(-centerX, -centerY), rotate and flip vertically,
    // scale, and translate(offsetCanvasX, offsetCanvasY)
    var centerX = scaleX * (viewBox[2] + viewBox[0]) / 2;
    var centerY = scaleY * (viewBox[3] + viewBox[1]) / 2;

    if (sideways) {
        this.transform = [0, scaleX, -scaleY, 0, 
                          offsetX + centerY + width / 2, 
                          offsetY - centerX + height / 2];
    } else {
        this.transform = [scaleX, 0, 0, scaleY,
                          offsetX - centerX + width / 2,
                          offsetY - centerY + height / 2];
    }

    this.width = width;
    this.height = height;
...
```